### DISCUSSION TITLE
Flutter ios build failed | cocoapods error

### DISCUSSION THREAD

**Discussion by ainayves** [codemagic-ci-cd/codemagic-docs#1682] [Q&A] [answered]
Hello all,
I was using Codemagic to publish apps on Appstore during more than 6 months , it was working perfectly even if I am using Ubuntu.
But now , I have this issue about cocapods . Could anyone help me please ? I am not very familiar with MacOs erros. Thank you very much .

`Upgrading project.pbxproj
Upgrading Runner.xcscheme
Upgrading Info.plist
Removing script build phase dependency analysis.
Adding input path to Thin Binary build phase.
Automatically signing iOS for device deployment using specified development team in Xcode project: 3239Q23S3A
Running pod install...                                             533ms
CocoaPods' output:
Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)
    [!] Unable to find a target named `RunnerTests` in project `Runner.xcodeproj`, did find `Runner`.
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer/target_inspector.rb:108:in `compute_targets'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer/target_inspector.rb:40:in `compute_results'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1197:in `block (3 levels) in inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1195:in `each'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1195:in `block (2 levels) in inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1193:in `each'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1193:in `block in inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1188:in `inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:107:in `analyze'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:422:in `analyze'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:244:in `block in resolve_dependencies'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:243:in `resolve_dependencies'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:162:in `install!'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/command/install.rb:52:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/command.rb:52:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/bin/pod:55:in `<top (required)>'
    /Users/builder/.rbenv/versions/3.0.4/bin/pod:25:in `load'
    /Users/builder/.rbenv/versions/3.0.4/bin/pod:25:in `<main>'
    
    Error output from CocoaPods:
    
    [!] Automatically assigning platform `iOS` with version `11.0` on target `Runner` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.

Error running pod install

Build failed :|
Failed to build for iOS
    
`

**Comment by ainayves** [comment]
@maks

Hello , I resolved the issue using your advice @maks , but I face another one now : 

`Automatically signing iOS for device deployment using specified development team in Xcode project: 3239Q23S3A
Running pod install...                                             596ms
CocoaPods' output:
↳
      Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)
      Using `ARCHS` setting to build architectures of target `Pods-RunnerTests`: (``)

    Fetching external sources
    -> Fetching podspec for `Flutter` from `Flutter`
    -> Fetching podspec for `connectivity_plus` from `.symlinks/plugins/connectivity_plus/ios`
    -> Fetching podspec for `fluttertoast` from `.symlinks/plugins/fluttertoast/ios`
    -> Fetching podspec for `geocoding_ios` from `.symlinks/plugins/geocoding_ios/ios`
    -> Fetching podspec for `geolocator_apple` from `.symlinks/plugins/geolocator_apple/ios`
    -> Fetching podspec for `image_picker_ios` from `.symlinks/plugins/image_picker_ios/ios`
    -> Fetching podspec for `libphonenumber_plugin` from `.symlinks/plugins/libphonenumber_plugin/ios`
    -> Fetching podspec for `path_provider_foundation` from `.symlinks/plugins/path_provider_foundation/darwin`
    -> Fetching podspec for `phone_number` from `.symlinks/plugins/phone_number/ios`
    -> Fetching podspec for `qr_code_scanner` from `.symlinks/plugins/qr_code_scanner/ios`
    -> Fetching podspec for `sqflite` from `.symlinks/plugins/sqflite/ios`

    Resolving dependencies of `Podfile`
      CDN: trunk Relative path: CocoaPods-version.yml exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_f_4_e.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/f/4/e/FMDB/2.7.5/FMDB.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_d_d_6.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/d/d/6/ReachabilitySwift/5.0.0/ReachabilitySwift.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_5_1_0.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/5/1/0/Toast/4.0.0/Toast.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_2_4_b.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/2/4/b/PhoneNumberKit/3.3.4/PhoneNumberKit.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_4_2_c.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/4/2/c/FlutterMacOS/3.0.0/FlutterMacOS.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_0_2_a.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/2/a/MTBBarcodeScanner/5.0.11/MTBBarcodeScanner.podspec.json exists! Returning local because checking is only performed in repo update
    [!] CocoaPods could not find compatible versions for pod "PhoneNumberKit/PhoneNumberKitCore":
      In Podfile:
        phone_number (from `.symlinks/plugins/phone_number/ios`) was resolved to 1.0.0, which depends on
          PhoneNumberKit/PhoneNumberKitCore (= 3.5.8)

    None of your spec sources contain a spec satisfying the dependency: `PhoneNumberKit/PhoneNumberKitCore (= 3.5.8)`.

    You have either:
     * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
     * mistyped the name or version.
     * not added the source repo that hosts the Podspec to your Podfile.

    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:317:in `raise_error_unless_state'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:299:in `block in unwind_for_conflict'
    <internal:kernel>:90:in `tap'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:297:in `unwind_for_conflict'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:257:in `process_topmost_state'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:182:in `resolve'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/molinillo-0.8.0/lib/molinillo/resolver.rb:43:in `resolve'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/resolver.rb:94:in `resolve'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1082:in `block in resolve_dependencies'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1080:in `resolve_dependencies'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:125:in `analyze'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:422:in `analyze'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:244:in `block in resolve_dependencies'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:243:in `resolve_dependencies'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:162:in `install!'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/command/install.rb:52:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/command.rb:52:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/bin/pod:55:in `<top (required)>'
    /Users/builder/.rbenv/versions/3.0.4/bin/pod:25:in `load'
    /Users/builder/.rbenv/versions/3.0.4/bin/pod:25:in `<main>'

Error output from CocoaPods:
↳

    [!] Automatically assigning platform `iOS` with version `11.0` on target `Runner` because no platform was specified. Please specify a platform for this target in your
    Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.

Error: CocoaPods's specs repository is too out-of-date to satisfy dependencies.
To update the CocoaPods specs, run:
  pod repo update



Error running pod install`

**Comment by martinjeret** [comment]
this was the previous discussion that helped the OP https://www.reddit.com/r/iosdev/comments/13l5m9k/comment/jkqwlnv/?context=3

**Comment by icarusdust** [comment]
Hi, the error was thrown due to unsynchronized dependency in the Podfile and the logs look self-explanatory by giving a solution:

```
You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile
```

Or just remove `Podfile.lock ` and let it re-create with the updated version of that dependency along with other dependencies

**Reply by ainayves** [reply]
Hi , I already did an `pod repo update` and `pod install --repo-update` but without success

**Reply by icarusdust** [reply]
What about the .lock file ?

**Reply by ainayves** [reply]
Ther is no .lock file her , wher should I find it please ?
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/66997516/f439344a-7566-450d-a1b9-0592c14ae2a4)

**Reply by ainayves** [reply]
There was not a lock file , I am already using cocoapods latest version in codemagic

**Reply by ainayves** [reply]
@martinjeret  @icarusdust  Is there a way to launch this command automatically when running the build ? Thanks .

`pod 'PhoneNumberKit', :git => 'https://github.com/marmelroy/PhoneNumberKit'`

**Reply by icarusdust** [reply]
@ainayves it is not a command but a framework for validating phone numbers and it is not about what Cocoapods version you are using in Codemagic. Check [that framework](https://github.com/marmelroy/PhoneNumberKit) for more info.

The framework can be installed in a few ways and as your project is Cocoapods based, adding `pod 'PhoneNumberKit', :git => 'https://github.com/marmelroy/PhoneNumberKit'` to the Podfile is the way needed. 

So, what is the issue here ? => As stated in the logs:

```
[!] CocoaPods could not find compatible versions for pod "PhoneNumberKit/PhoneNumberKitCore":
  In Podfile:
    phone_number (from `.symlinks/plugins/phone_number/ios`) was resolved to 1.0.0, which depends on
      PhoneNumberKit/PhoneNumberKitCore (= 3.5.8)

None of your spec sources contain a spec satisfying the dependency: `PhoneNumberKit/PhoneNumberKitCore (= 3.5.8)`.
```
How to solve it ? 
1. Check the framework in the Podfile (their official doc i shared above says that it needs to be installed by adding it in the Podfile along with its version in front of it which is `3.5.10` => but you can still use `3.5.8` because judging from the error,  `phone_number`  depends on `3.5.8`, so it should suffice ) ( FYI Podfile.lock is generated after running `pod install `successfully and the file holds pods along with their versions used in the project. I strongly suggest you to experiment it by running `pod install `locally to see the results)
2. If the framework already exists, then you need to upgrade/downgrade its version to the required version in the error log
3. Check if the line `platform :ios, '14.0'` is present in the Podfile and uncommented (if the ios version is less than 11, upgrade it)

Coming back to your question if that link can be added automatically, the answer is a no really because eventually it is up to you as a developer who needs to have that framework in the project which means that it is up to you to add or remove it. 

Hope it makes sense.

**Reply by ainayves** [reply, marked as answer]
Thank you ! I just removed the 'phone_number' package and the build was running fine.
